### DISCUSSION TITLE
Help with Nested.Join

### DISCUSSION THREAD

**Discussion by jabanks2** [TabularEditor/TabularEditor3#1316] [Data Modeling]
I'm modeling a report with TB3 and moving it to SSAS.  When I create Tables in TB3 using power query from PBI.  The Nested.Joins are not working as expected. 

Here is the code from Power Query: 
#"Merged DivGroupMgr" = Table.NestedJoin(#"Added Location Type", {"DivisionGroupManager"}, Employees, {"D_Employeesid"}, "Dim_Employees", JoinKind.LeftOuter), 

Once copied to TB3 model... I get an error when I try to refresh data. 
THE JSON DDL request failed with the following error: <ccon> Failed to save modifications to the server. Error returned: 'Expression in partition 'Branches' in table 'Branches' references unknown variable or function 'Employees'.'.</ccon>.

I tried creating Relationships between the 2 tables and that didn't solve my issue. 

I searched the web and found no answers.  Any help would be greatly appreciated. 

Thanks.

**Comment by jabanks2** [comment]
![image](https://github.com/TabularEditor/TabularEditor3/assets/82853034/28379d11-90ce-404f-8a87-e2688a5a9220)

**Reply by edhans** [reply]
It is saying there is no Employees table to join with. 
It might be you have Employees loaded. I never join with loaded tables, I always create a shared expression just for the join. I am not even sure if you can join with a loaded Table in TE/3. I've never tried. Shared Expressions only.

**Reply by jabanks2** [reply]
I'll clear the table to see if that works.  Thanks

**Reply by jabanks2** [reply]
It didn't work...

**Reply by edhans** [reply]
I don't know what didn't work. I am not sure why you said you were going to clear the table. That won't do anything.

Is your employee table set to load to the model? Or is it a shared expression?
